### PR TITLE
Refactor product model

### LINKED ISSUE #1497: Refactor product model to prepare for featured pages
Featured product pages will introduce more logic to the product model, so before adding that logic let's take this opportunity to re-organize the code a little better

### PR DESCRIPTION
**Pull request by dhmacs**
closes #1497 

The goal of this PR is to clean up and re-organize the product model code in preparation for additional featured product page features.

## QA

Perform a smoke test on product pages

### CODE REVIEW

File: frontend/templates/product/sections/CrossSellSection/useAvalableForSaleVariants.tsx
Comment (masonmcelvain) [inline frontend/templates/product/sections/CrossSellSection/useAvalableForSaleVariants.tsx:1]: Typo: this file should be named `useAvailableForSaleVariants.tsx`, not `useAvalableForSaleVariants.tsx`

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Nice work!

File: frontend/models/product/components/product-variant.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+   const { crossSell, ...other } = variant;
+   const isDiscounted =
+      variant.compareAtPrice != null &&
+      parseFloat(variant.compareAtPrice.amount) >
+         parseFloat(variant.price.amount);
+   const discountPercentage = isDiscounted
+      ? computeDiscountPercentage(
+           parseFloat(variant.price.amount) * 100,
+           parseFloat(variant.compareAtPrice!.amount) * 100
+        )
```
[the remark this answers is not in the document]
Comment (dhmacs) [reply frontend/models/product/components/product-variant.ts:79]: Good suggestion 👍 I can change this in the next PR

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

File: frontend/templates/product/sections/CrossSellSection/useAvalableForSaleVariants.tsx
Comment (masonmcelvain) [reply frontend/templates/product/sections/CrossSellSection/useAvalableForSaleVariants.tsx:1]: Yep!

**Comment by deltuh-vee** [comment]
QA 🎬 
Product Pages are still working.

**Pull request merged**